### DISCUSSION TITLE
Id of a new record

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#1698] [General]
Hi, how can i get the id of a new record inserted in the database, thanks in advance

**Comment by danharrin** [comment]
`$this->record->id`?

**Reply by jcgmalave** [reply]
i get this message: Property [$record] not found on component:

**Reply by danharrin** [reply]
You gave no information so its hard for me to recommend an accurate solution.

**Comment by jcgmalave** [comment]
Thanks a lot for the quick response, what im try to do is this, i need to update a field from my Product Model, bur a i need to do it after i register a record in one of the HasMany relationships, for example, i have Product with the field quantity, and i have Product Detail, after add one product detail i need to update the master

**Reply by danharrin** [reply]
Not sure I understand, can you send code and tell me where you want to access the record ID with a comment?

**Reply by jcgmalave** [reply]
i need to update the product quantity  according to the result of ProductSizeRelationManager extends HasManyRelationManager

$product = Product::where('id',$this->record->product_id)->first();
$product->quantity = 777;
$product->save();

**Reply by danharrin** [reply]
Where is this code in your relation manager though?

**Reply by jcgmalave** [reply]
Yes, i have a master record which is Product and a Detail Page, ProducDetails, i need to update the quantity field in the Product Model, while the ProductDetail is being generated

**Reply by danharrin** [reply]
Please send me the code for your relation manager.

**Reply by jcgmalave** [reply]
<?php

namespace App\Filament\Resources\ProductResource\RelationManagers;

use App\Models\Product;
use App\Models\ProductSize;
use App\Models\Size;
use App\Models\SizeType;
use App\Models\User;
use Filament\Forms;
use Filament\Resources\Form;
use Filament\Resources\RelationManagers\HasManyRelationManager;
use Filament\Resources\Table;
use Filament\Tables;
use Illuminate\Support\Facades\Auth;

class ProductSizeRelationManager extends HasManyRelationManager
{
    protected static string $relationship = 'ProductSizes';
    protected static ?string $label = 'Detalle de Productos';
    protected static ?string $recordTitleAttribute = 'title';
    

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\Select::make('size_type_id')
                ->label('Tipo Talle')
                ->options(SizeType::all()->pluck('name', 'id'))
                ->reactive()
                ->required()
                ->afterStateUpdated(fn (callable $set) => $set('size_id',null)),

                Forms\Components\Select::make('size_id')
                ->label('Talle')
                ->required()
                ->options(function(callable $get){
                    
                    $size_type = SizeType::find($get('size_type_id'));

                    if(!$size_type){
                        return Size::all()->pluck('name', 'id')->toArray();
                    }else{
                        //$size_type->sizes->pluck(value: 'name', key: 'id')->toArray();
                        return Size::where('size_type_id',$get('size_type_id'))->pluck('name', 'id')->toArray();
                    }

                    
                }),
                
                /*Forms\Components\BelongsToSelect::make('size_id')
                    ->relationship('size', 'name')->required(),*/
                
                Forms\Components\BelongsToSelect::make('color_id')
                    ->relationship('color', 'name')
                    ->label('Color')
                    ->required(),
                Forms\Components\TextInput::make('quantity')
                    ->label('Cantidad')
                    ->required()
                    ->numeric(),
                Forms\Components\FileUpload::make('image')->disk('public')
                    ->directory('products/colors')
                    ->visibility('public')
                    ->label('Imagen')
                    //->imagePreviewHeight('200')
                    //->preserveFilenames()
                    ->required(),    
                Forms\Components\Select::make('user_id')
                    ->label('Usuario')
                    ->options(User::where('id',Auth::user()->id)->pluck('name', 'id'))
                    ->default(Auth::user()->id)
                    ->disablePlaceholderSelection()
                    ->required(),    
            ]);
    }

    public static function table(Table $table): Table
    {
        
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('size.name')->label('Talle'),
                Tables\Columns\TextColumn::make('color.name')->label('Color'),
                Tables\Columns\TextColumn::make('quantity')->label('Cantidad'),
                Tables\Columns\ImageColumn::make('image')->label('Imagen')
                ->url(fn (ProductSize $record): string => route('product.details', ['slug' => $record->product->slug]))
                ->openUrlInNewTab(),
            ])
            ->filters([
                //
            ]);
    }

    protected function afterCreate(): void
    {
        
        //$product_size = ProductSize::where('id',$this->record->id)->first();
        
        $product = Product::where('id',1)->first();
        $product->quantity = 777;
        $product->save();

    }
}
